src/components/VlanForm.js

```diff
@@ -0,0 +1,152 @@
+import cockpit from 'cockpit';
+import { useNetworkDispatch, useNetworkState, addConnection, updateConnection } from '../NetworkContext';
+import interfaceType from '../lib/model/interfaceType';
+
+const _ = cockpit.gettext;
+
+const VlanForm = ({ isOpen, onClose, connection }) => {
+    const { vlan } = connection || {};
+    const isEditing = !!connection;
+    const [name, setName] = useState(connection?.name);
+    const [vlanId, setVlanId] = useState(vlan?.vlanId || 0);
+    const [parentDevice, setParentDevice] = useState(vlan?.parentDevice);
+    const [candidateInterfaces, setCandidateInterfaces] = useState([]);
+    const { interfaces } = useNetworkState();
+    const dispatch = useNetworkDispatch();
+    const [suggestName, setSuggestName] = useState(!isEditing);
+
+    useEffect(() => {
+        setCandidateInterfaces([...Object.values(interfaces)]);
+        if (!parentDevice) setParentDevice(Object.values(interfaces)[0]?.name);
+    }, [interfaces]);
+
+    useEffect(() => {
+        proposeName();
+    }, [vlanId, parentDevice]);
+
+    const addOrUpdateConnection = () => {
+        let promise = null;
+
+        if (isEditing) {
+            promise = updateConnection(dispatch, connection, { vlan: { name, vlanId, parentDevice } });
+        } else {
+            promise = addConnection(dispatch, { name, type: interfaceType.VLAN, vlan: { vlanId, parentDevice } });
+        }
+
+        promise.then(onClose).catch(console.error);
+    };
+
+    const isIncomplete = () => {
+        if (name === "") return true;
+        if (vlanId === "") return true;
+        if (parentDevice === "") return true;
+
+        return false;
+    };
+
+    const updateName = (value) => {
+        setName(value);
+        setSuggestName(false);
+    };
+
+    const proposeName = () => {
+        if (!suggestName) return;
+        setName(`${parentDevice}.${vlanId}`);
+    };
+
+    if (!parentDevice) return null;
+
+    return (
+        <Modal
+            variant={ModalVariant.small}
+            title={isEditing ? _("Edit VLAN") : _("Add VLAN")}
+            isOpen={isOpen}
+            onClose={onClose}
+            actions={[
+                <Button key="confirm" variant="primary" onClick={addOrUpdateConnection} isDisabled={isIncomplete()}>
+                    {isEditing ? _("Change") : _("Add")}
+                </Button>,
+                <Button key="cancel" variant="link" onClick={onClose}>
+                    {_("Cancel")}
+                </Button>
+            ]}
+        >
+            <Form>
+                <FormGroup
+                    label={_("Parent")}
+                    isRequired
+                    fieldId="parentDevice"
+                >
+
+                    <FormSelect value={parentDevice} onChange={setParentDevice} id="parentDevice">
+                        {candidateInterfaces.map(({ name }, index) => (
+                            <FormSelectOption key={index} value={name} label={name} />
```

Comment:
I think it would be better to use the name (which should be unique, shouldn't it?) as key too.